### DISCUSSION TITLE
ImportError: cannot import name 'NiftiDataset' from 'monai.data' (/usr/local/lib/python3.7/dist-packages/monai/data/__init__.py)

### DISCUSSION THREAD

**Discussion by Sebagam** [Project-MONAI/MONAI#2934] [Q&A]
Hi all,

When trying to run MONAI on google colab:

ImportError: cannot import name 'NiftiDataset' from 'monai.data' (/usr/local/lib/python3.7/dist-packages/monai/data/__init__.py)

This issue was not present a few months ago.

Best,
Sebastian

**Comment by Nic-Ma** [comment]
Hi @Sebagam ,

Thanks for your interest here.
We already changed it to `ImageDataset` in v0.5.

Thanks.

**Comment by Sebagam** [comment]
Ok, thanks.
It looks that I have other compatibility issues.
How can I install v0.4?

**Comment by Nic-Ma** [comment]
Maybe you can try pip install monai==0.4.0?

Thanks.

**Comment by Nic-Ma** [comment]
But I strongly suggest you to update with MONAI v0.6 as we integrated many enhancements.

Thanks.

**Comment by Nic-Ma** [comment]
Hi @Sebagam ,

If you check the error log, you can find that the data shape is wrong:
Shape: (1, 166, 256, 256, 1)
Actually your data already has channel dim at the last dim, you should not use `AddChannel` transform, you can use `EnsureChannelFirst` or `AsChannelFirst` transform.

Thanks.

**Comment by Sebagam** [comment]
Thank you, Nic,

Do you know what the following issue mean?

RuntimeError: DataLoader worker (pid(s) 1038) exited unexpectedly

**Comment by Nic-Ma** [comment]
Hi @Sebagam ,

I think maybe it's some PyTorch multi-processing issue? Can you try again with `num_workers=0`?

Thanks.

**Comment by Nic-Ma** [comment]
Hi @Sebagam ,

Seems something wrong with your data shape?
Could you please help add `EnsureChannelFirst` transform at the beginning of your transform chain?

Thanks.

**Comment by Sebagam** [comment]
I got this error now:
ValueError: meta_dict not available, EnsureChannelFirst is not in use.

**Comment by Nic-Ma** [comment]
Hi @Sebagam ,

As you are using array-based transforms instead of dict-based transforms, it may not work well.
Could you please help change to `AsChannelFirst` transform?

Thanks.

**Reply by pip-alireza** [reply]
Tried running the Swin-unet-r on BTCV data, and it keeps giving the metadata not available error. I used the code provided for BTCV data, do you know what is causing this? (checked the metadata and it is available. Adding the EnsureChannelFirst/AsChannelFirst did not help but caused the caching dataset to fail.) 

Here is the error:
File "/home/alireza/anaconda3/envs/pytorch2/lib/python3.11/site-packages/monai/transforms/utility/array.py", line 204, in __call__
    raise ValueError(msg)
ValueError: Metadata not available and channel_dim=None, EnsureChannelFirst is not in use.

**Reply by KumoLiu** [reply]
Hi @pip-alireza, you could specify the channel_dim in `EnsureChannelFirst`.
https://github.com/Project-MONAI/MONAI/blob/90d2acb54d5325e6727f2cc2329cb40adc8577af/monai/transforms/utility/array.py#L180-L183
Something like:
```
EnsureChannelFirst(channel_dim="no_channel")
```

Hope it helps, thanks!

**Reply by preet012** [reply]
I was facing the same issue. Adding `EnsureChannelFirstd(keys=["image", "label"],channel_dim='no_channel'),` right after LoadImaged in transforms helped.

**Reply by mtoan65** [reply]
@preet012 Can you send the details of that revised code snippet here?

**Comment by Nic-Ma** [comment]
I think maybe you can print out the model predictions, see whether it's overfitting?
@wyli @ericspod Could you guys help please share some comments about the training status?

Thanks in advance.

**Comment by Sebagam** [comment]
I think that I managed to fix it.
Do you know how to turn off this output?

INFO:nibabel.global:pixdim[0] (qfac) should be 1 (default) or -1;
setting qfac to 1

/usr/local/lib/python3.7/dist-packages/monai/data/utils.py:542:
UserWarning: Modifying image pixdim from [1.1999999 1.        1.
 0.       ] to [  1.19999993   1.           1.         218.40411641]
  warnings.warn(f"Modifying image pixdim from {pixdim} to {norm}")



Thanks

**Comment by Nic-Ma** [comment]
I think you can turn off all the warnings, referring to:
https://stackoverflow.com/questions/14463277/how-to-disable-python-warnings

Thanks.

**Comment by Sebagam** [comment]
Hi Nic,

Yes, I know that command.
The problem is the Nibabel information messages:

INFO:nibabel.global:pixdim[0] (qfac) should be 1 (default) or -1;
setting qfac to 1

pixdim[0] (qfac) should be 1 (default) or -1; setting qfac to 1



Do you know how to suppress them?


Thanks

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Do you know something about this Nibabel warning?

Thanks in advance.
